lib/active_merchant/billing/gateways/merchant_e_solutions.rb

```diff
@@ -27,8 +27,10 @@ def initialize(options = {})
 
       def authorize(money, creditcard_or_card_id, options = {})
         post = {}
-        post[:client_reference_number] = options[:customer] if options.has_key?(:customer)
-        post[:moto_ecommerce_ind] = options[:moto_ecommerce_ind] if options[:moto_ecommerce_ind]
+        post[:client_reference_number] = options[:customer] if options[:customer].present?
+        post[:moto_ecommerce_ind] = options[:moto_ecommerce_ind] if options[:moto_ecommerce_ind].present?
+        post[:recurring_pmt_num] = options[:recurring_pmt_num] if options[:recurring_pmt_num].present?
+        post[:recurring_pmt_count] = options[:recurring_pmt_count] if options[:recurring_pmt_count].present?
```

Comment:
You can remove the `.present?` check here. The implicit check is enough for our purpose.